### PR TITLE
DOP-3792: added featureFlagSearchUI

### PR DESCRIPTION
**Pull request by mmeigs**
Adds feature flag for Search UI Improvements project.

It is piped through to the frontend. Most likely will be utilized in Snooty, but entire implementation still undecided.

AWS Parameter Store values can be found [here](https://us-east-2.console.aws.amazon.com/systems-manager/parameters/?region=us-east-2&tab=Table#list_parameter_filters=Name:Contains:search_ui). For now, all are false. Staging will be changed to true at some point in near future.

### CODE REVIEW

File: src/job/jobHandler.ts [hunk 1]
```diff
@@ -342,6 +342,7 @@ export abstract class JobHandler {
     const snootyFrontEndVars = {
       GATSBY_BASE_URL: this._config.get<string>('gatsbyBaseUrl'),
       PREVIEW_BUILD_ENABLED: this._config.get<string>('previewBuildEnabled'),
+      FEATURE_FLAG_SEARCH_UI: this._config.get<string>('featureFlagSearchUI'),
```
Comment (seungpark) [inline src/job/jobHandler.ts:345]: minor nitpick. would prefix with `GATSBY_` but not a deal breaker

**Review by seungpark — APPROVED** [review]
this LGTM besides a minor naming nitpick!

File: src/job/jobHandler.ts (hunk 1 above)
Comment (mmeigs) [reply src/job/jobHandler.ts:345]: ahhh, I see this in the documentation now. I think that's reasonable. How do we feel about this (FE) being `GATSBY_TEST_SEARCH_UI` while the rest of the pipeline of the variable (BE) stays `FEATURE_FLAG_SEARCH_UI`? Should I change all of the naming to match?

Comment (mmeigs) [reply src/job/jobHandler.ts:345]: Oh, sure! https://www.gatsbyjs.com/docs/how-to/local-development/environment-variables/

**Review by seungpark — APPROVED** [review]
this LGTM 👍

**Pull request merged**